File: day_02/part_1/solution.js
const fs = require("fs");

let safeCount = 0;

const fileContent = fs.readFileSync("../input.txt", "utf8");
const lines = fileContent.trim().split("\n");

for (let line of lines) {
    const report = line.split(" ").map(Number);

    let increasing = true;
    let decreasing = true;
    let validDifferences = true;

    for (let i = 0; i < report.length - 1; i++) {
        if (report[i] > report[i + 1]) {
            increasing = false;
        }
        if (report[i] < report[i + 1]) {
            decreasing = false;
        }
        if (!(1 <= Math.abs(report[i] - report[i + 1]) && Math.abs(report[i] - report[i + 1]) <= 3)) {
            validDifferences = false;
        }
    }

    if ((increasing || decreasing) && validDifferences) {
        safeCount += 1;
    }
}

console.log(`Answer: ${safeCount}`);

File: day_02/part_1/solution.py
safe_count = 0

with open("../input.txt", 'r') as file:
    for line in file:
        report = list(map(int, line.strip().split()))
        
        increasing = True
        decreasing = True
        valid_differences = True

        for i in range(len(report) - 1):
            if report[i] > report[i + 1]:
                increasing = False
            if report[i] < report[i + 1]:
                decreasing = False
            if not (1 <= abs(report[i] - report[i + 1]) <= 3):
                valid_differences = False

        if (increasing or decreasing) and valid_differences:
            safe_count += 1

print(f"Answer: {safe_count}")